diff --git aircrack-ng-1.2-beta2/common.mak-orig aircrack-ng-1.2-beta2/common.mak
index 8c6a3ef..f1cbb6e 100644
--- aircrack-ng-1.2-beta2/common.mak-orig
+++ aircrack-ng-1.2-beta2/common.mak
@@ -31,7 +31,7 @@ COMMON_CFLAGS	=
 
 
 ifeq ($(subst TRUE,true,$(filter TRUE true,$(sqlite) $(SQLITE))),true)
-	COMMON_CFLAGS	+= -I/usr/local/include -DHAVE_SQLITE
+	COMMON_CFLAGS	+= -DHAVE_SQLITE
 endif
 
 ifeq ($(OSNAME), cygwin)
