From 2b742504845c52b6000aba08eff23c08096ece9c Mon Sep 17 00:00:00 2001
From: Scott Ellis <scott@jumpnowtek.com>
Date: Mon, 12 Sep 2016 12:18:50 -0400
Subject: [PATCH 2/2] spidev: Add a generic compatible id

---
 drivers/spi/spidev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index 409a8c5..81c1e1e 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -684,6 +684,7 @@ static int spidev_remove(struct spi_device *spi)
 
 static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "rohm,dh2228fv" },
+	{ .compatible = "generic,spi" },
 	{},
 };
 
-- 
2.7.4

