diff --git git/include/configs/omap4_common.h-orig git/include/configs/omap4_common.h
index 1ea4973..7d74080 100644
--- git/include/configs/omap4_common.h-orig
+++ git/include/configs/omap4_common.h
@@ -134,11 +134,13 @@
 	"loadaddr=0x82000000\0" \
 	"console=ttyO2,115200n8\0" \
 	"vram=16M\0" \
+	"coherent_pool=1M\0" \
 	"mmcdev=0\0" \
 	"mmcroot=/dev/mmcblk0p2 rw\0" \
 	"mmcrootfstype=ext3 rootwait\0" \
 	"mmcargs=setenv bootargs console=${console} " \
 		"vram=${vram} " \
+		"coherent_pool=${coherent_pool} " \
 		"root=${mmcroot} " \
 		"rootfstype=${mmcrootfstype}\0" \
 	"loadbootscript=fatload mmc ${mmcdev} ${loadaddr} boot.scr\0" \
