From 151460d4338f96e81d1fb8d4d5bb4fabc11029f9 Mon Sep 17 00:00:00 2001
From: Scott Ellis <scott@jumpnowtek.com>
Date: Sat, 26 Apr 2014 16:08:40 -0400
Subject: [PATCH] revert a6cc060 breaks duovero

---
 src/drivers/driver_nl80211.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git git/src/drivers/driver_nl80211.c git/src/drivers/driver_nl80211.c
index 1300703..ef94f24 100644
--- git/src/drivers/driver_nl80211.c
+++ git/src/drivers/driver_nl80211.c
@@ -8184,10 +8184,6 @@ static int nl80211_setup_ap(struct i802_bss *bss)
 		if (nl80211_mgmt_subscribe_ap(bss))
 			return -1;
 
-	if (drv->device_ap_sme && !drv->use_monitor)
-		if (nl80211_mgmt_subscribe_ap_dev_sme(bss))
-			return -1;
-
 	if (!drv->device_ap_sme && drv->use_monitor &&
 	    nl80211_create_monitor_interface(drv) &&
 	    !drv->device_ap_sme)
-- 
1.8.3.2

