From 3dde377d5950052dcb723ba0d8667308c7c1a25a Mon Sep 17 00:00:00 2001
From: Scott Ellis <scott@jumpnowtek.com>
Date: Sat, 11 Mar 2017 13:32:48 -0500
Subject: [PATCH 4/4] rtc-twl: Enable battery charging

---
 drivers/rtc/rtc-twl.c | 39 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/drivers/rtc/rtc-twl.c b/drivers/rtc/rtc-twl.c
index 2dc787d..7f522b3 100644
--- a/drivers/rtc/rtc-twl.c
+++ b/drivers/rtc/rtc-twl.c
@@ -470,6 +470,41 @@ static struct rtc_class_ops twl_rtc_ops = {
 	.alarm_irq_enable = twl_rtc_alarm_irq_enable,
 };
 
+#define REG_BBSPOR_CFG 0xE6
+#define VRTC_EN_SLP_STS        (1 << 6)
+#define VRTC_EN_OFF_STS        (1 << 5)
+#define VRTC_PWEN      (1 << 4)
+#define BB_CHG_EN      (1 << 3)
+#define BB_SEL_1       (1 << 2)
+#define BB_SEL_0       (1 << 1)
+
+static int enable_rtc_battery_charging(void)
+{
+	int ret;
+	u8 data;
+
+	ret = twl_i2c_read_u8(TWL6030_MODULE_ID0, &data, REG_BBSPOR_CFG);
+	if (ret < 0) {
+		pr_err("read bbspor_cfg failed: %d\n", ret);
+		return ret;
+	}
+
+	/*
+	 * Charge battery to 3.15v
+	 * TWL6030 Register Map, Table 224, BBSPOR_CFG Register
+	 */
+	data &= ~BB_SEL_0;
+	data |= (BB_SEL_1 | BB_CHG_EN);
+
+	ret = twl_i2c_write_u8(TWL6030_MODULE_ID0, data, REG_BBSPOR_CFG);
+	if (ret < 0)
+		pr_err("write bbspor_cfg failed: %d\n", ret);
+	else
+		pr_info("enabled rtc battery charging\n");
+
+	return ret;
+}
+
 /*----------------------------------------------------------------------*/
 
 static int twl_rtc_probe(struct platform_device *pdev)
@@ -525,6 +560,10 @@ static int twl_rtc_probe(struct platform_device *pdev)
 	if (ret < 0)
 		return ret;
 
+	ret = enable_rtc_battery_charging();
+	if (ret < 0)
+		dev_err(&pdev->dev, "Failed to enable rtc battery charging\n");
+
 	device_init_wakeup(&pdev->dev, 1);
 
 	rtc = devm_rtc_device_register(&pdev->dev, pdev->name,
-- 
2.7.4

